test/jdk/tools/pack200/PackTestZip64.java

```diff
@@ -85,43 +91,62 @@ static void testPacking() throws IOException {
         cmdsList.add(unpackedFile.getName());
         Utils.runExec(cmdsList);
 
-        // Compare files binary
-        compareTwoFiles(testFile, unpackedFile);
+        if (!testFile.exists()) {
+            throw new IOException("File " + testFile.getName() + " does not exist!");
+        }
+
+        if (!unpackedFile.exists()) {
+            throw new IOException("File " + unpackedFile.getName() + " does not exist!");
+        }
+        // Compare two Jar files
+        compareTwoJarFiles(new JarFile(testFile), new JarFile(unpackedFile));
 
         // Cleaning up generated files
         testFile.delete();
         packedFile.delete();
         unpackedFile.delete();
     }
 
-    static void compareTwoFiles(File src, File dst) throws IOException {
-        if (!src.exists()) {
-            throw new IOException("File " + src.getName() + " does not exist!");
-        }
-
-        if(!dst.exists()) {
-            throw new IOException("File " + dst.getName() + " does not exist!");
-        }
+    static void compareTwoJarFiles(JarFile srcJar, JarFile dstJar) throws IOException {
+        for (Enumeration srcEntries = srcJar.entries(); srcEntries.hasMoreElements(); ) {
+            JarEntry srcEntry = (JarEntry) srcEntries.nextElement();
+            JarEntry dstEntry = dstJar.getJarEntry(srcEntry.getName());
+            if (dstEntry == null) {
+                throw new IOException("Jar Entry "+srcEntry.getName()+" does not exist in "+dstJar.getName());
+            }
 
-        BufferedInputStream srcis, dstis;
-        srcis = new BufferedInputStream(new FileInputStream(src));
-        dstis = new BufferedInputStream(new FileInputStream(dst));
+            BufferedInputStream srcis = new BufferedInputStream(srcJar.getInputStream(srcEntry);
+            BufferedInputStream dstis = new BufferedInputStream(dstJar.getInputStream(dstEntry);
 
-        int s = 0, d, pos = 0;
-        while (s != -1) { // Checking of just one result for EOF is enough
-            s = srcis.read();
-            d = dstis.read();
+            for (int pos=0; ;++pos) {
+                int s = srcis.read();
+                int d = dstis.read();
 
-            if (s != d) {
-                throw new IOException("Files are differ starting at position: "
-                + Integer.toHexString(pos));
+                if (s != d) {
+                    throw new IOException("Files differ starting at position: "
```

Comment:
Perhaps you missed it, but I suggested adding "0x" at the end of this string to make it clear the position is expressed in hex. Please add it.